File: code/ch15/15.2.1.jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        bool dp[nums.size()];
        memset(dp, false, sizeof(dp));
        dp[0] = true;
        for (int i = 1; i < nums.size(); i++) {
            for (int j = 0; j < i; j++) {
                if (j + nums[j] >= i) {
                    dp[i] = dp[i] | dp[j];
                }
            }
        }
        return dp[nums.size() - 1];
    }
};

File: code/ch15/15.2.1.jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        dp = [False] * len(nums)
        dp[0] = True
        for i in range(1, len(nums)):
            for j in range(0, i):
                if j + nums[j] >= i:
                    dp[i] = dp[i] | dp[j]
        return dp[len(nums) - 1]
